db/main.py

```diff
@@ -0,0 +1,51 @@
+from fastapi import FastAPI, Depends, HTTPException
+from sqlalchemy.orm import Session
+
+from db import schemas, crud
+from db.engine import SessionLocal
+
+
+app = FastAPI()
+
+def get_db() -> Session:
+    db = SessionLocal()
+
+    try:
+        yield db
+    finally:
+        db.close()
+
+
+@app.get("/authors/", response_model=list[schemas.Author])
+def read_authors(skip: int = 0, limit: int = 10, db: Session = Depends(get_db)):
+    return crud.get_all_authors(db=db, skip=skip, limit=limit)
+
+
+@app.get("/authors/{author_id}", response_model=schemas.Author)
+def read_single_author(author_id: int, db: Session = Depends(get_db)):
+    db_author = crud.get_author(db, author_id=author_id)
+    if db_author is None:
+        raise HTTPException(status_code=404, detail="Author not found")
+    return db_author
+
+
+@app.post("/author/create/", response_model=schemas.Author)
+def create_author(author: schemas.AuthorCreate, db: Session = Depends(get_db)):
+    return crud.create_author(author=author, db=db)
```

Comment:
Consider adding error handling for the `create_author` endpoint to manage potential exceptions during database operations. This will help in providing more informative feedback to the client.